azurerm/resource_arm_netapp_snapshot_test.go

```diff
@@ -0,0 +1,340 @@
+package azurerm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMNetAppSnapshot_basic(t *testing.T) {
+	resourceName := "azurerm_netapp_snapshot.test"
+	ri := tf.AccRandTimeInt()
+	location := testLocation()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMNetAppSnapshotDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMNetAppSnapshot_basic(ri, location),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMNetAppSnapshotExists(resourceName),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAzureRMNetAppSnapshot_requiresImport(t *testing.T) {
+	if !features.ShouldResourcesBeImported() {
+		t.Skip("Skipping since resources aren't required to be imported")
+		return
+	}
+
+	resourceName := "azurerm_netapp_snapshot.test"
+	ri := tf.AccRandTimeInt()
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMNetAppSnapshotDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMNetAppSnapshot_basic(ri, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMNetAppSnapshotExists(resourceName),
+				),
+			},
+			{
+				Config:      testAccAzureRMNetAppSnapshot_requiresImport(ri, testLocation()),
+				ExpectError: testRequiresImportError("azurerm_netapp_snapshot"),
+			},
```

Comment:
and this can become - `data.RequiresImportErrorStep(testAccAzureRMNetAppSnapshot_requiresImport),`

Reply:
fixed